Lesson10/src/Shop.java

```diff
@@ -0,0 +1,82 @@
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Shop {
+    List<Product> shop = new LinkedList<>();
+    Scanner scanner = new Scanner(System.in);
+
+    public List<Product> getShop() {
+        return shop;
+    }
+
+    public void sortPrice(Shop shop) {
+        List<Product> sortListPrice = new LinkedList<>(shop.getShop());
+        sortListPrice.sort(Comparator.comparing(Product::getPrice).reversed());
+        System.out.println("Список всех товаров, сортированный по цене:");
+        for (Product item : sortListPrice) {
+            System.out.println(item.getName() + ". Цена: " + item.getPrice());
+        }
+    }
+
+    public void addProduct(Product product) throws ProductAlreadyExist {
+        for (int i = 0; i < shop.size(); i++) {
+            if (product.getId() == shop.get(i).getId()) {
+                throw new ProductAlreadyExist("Товар с таким id уже существует");
+            } else shop.add(product);
+            break;
+        }
+    }
+
+    public Shop showAllProducts(Shop shop) {
+        System.out.println("Список всех товаров: ");
+        for (Product product : shop.getShop()) {
+            System.out.println(product.getName());
+        }
+        return shop;
+    }
+
+    public void removeProduct() throws InvalidProductId {
+        System.out.println("Введите id товара, который хотите удалить: ");
+        int id = scanner.nextInt();
```

Comment:
please do some refactoring. Remove all work with scanner from this class, instead of it pass information that you need to work with as a method params, like
public void removeProduct(int id)
public void editProduct(Product p)

and so on